### PR TITLE
Targeting Datums Renamed (and global)

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

[Implements the backend required to make targeting datums global](https://github.com/tgstation/tgstation/commit/6901ead12e419530b7f646ea21094d4432d7385e)

It's inconsistent with the rest of basic ai for these to have a high degree of state, plus like, such a waste yaknow?

[Implements GET_TARGETING_STRATEGY](https://github.com/tgstation/tgstation/commit/d79c29134d03424a9f8bacd64c08cb41775fe8c0)

Regexes used:
new.*(/datum/targetting_datum[^,(]*)\(*\)* -> GET_TARGETING_STRATEGY($1)

Renamed all instances of targetting to targeting (also targetting datum -> targeting strategy)

I've used GET_TARGETING_STRATEGY at the source where the keys are actually used, rather then in the listing. This works out just fine.

## Why It's Good For The Game

Not a misspelled name through the whole codebase, very slightly less memory load for basically no downside (slight cpu cost maybe but not a significant one.

### CODE REVIEW

File: code/__DEFINES/ai/ventcrawling.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /// Key that holds a vent that we want to exit out of (when we're already in a pipenet)
 #define BB_EXIT_VENT_TARGET "BB_exit_vent_target"
 /// Do we plan on going inside a vent? Boolean.
-#define BB_CURRENTLY_TARGETTING_VENT "BB_currently_targetting_vent"
+#define BB_CURRENTLY_TARGETING_VENT "BB_currently_targetting_vent"
```
Comment (JohnFulpWillard) [inline code/__DEFINES/ai/ventcrawling.dm:6]:
```suggestion
#define BB_CURRENTLY_TARGETING_VENT "BB_currently_targeting_vent"
```
:trollface:

**Comment by LemonInTheDark** [comment]
Need to do something about the global getter thing. might just postprocess blackboards on init idk

File: code/__DEFINES/ai/ai_blackboard.dm [hunk 2]
```diff
@@ -43,11 +43,11 @@
 ///Targetting subtrees
 #define BB_BASIC_MOB_CURRENT_TARGET "BB_basic_current_target"
 #define BB_BASIC_MOB_CURRENT_TARGET_HIDING_LOCATION "BB_basic_current_target_hiding_location"
-#define BB_TARGETTING_DATUM "targetting_datum"
+#define BB_targeting_strategy "targeting_strategy"
```
Comment (MrMelbert) [inline code/__DEFINES/ai/ai_blackboard.dm:46]: Did you intend for these defines to be partially lowercase or did they get caught in the regex

Comment (LemonInTheDark) [reply code/__DEFINES/ai/ai_blackboard.dm:46]: caught in regex I fucked it all up

Comment (LemonInTheDark) [reply code/__DEFINES/ai/ai_blackboard.dm]: handled

**Comment by Jacquerel** [comment]
code looks good to me, it is just tripping up on a mob which was presumably refactored during the lifespan of this PR

**Pull request merged**